frontends/concrete-python/examples/prime-match/prime-match.py

```diff
@@ -0,0 +1,301 @@
+
+    def __repr__(self):
+        return f"{repr(self.order_type)} {self.order_quantity} of {repr(self.order_symbol)}"
+
+    @staticmethod
+    def random(skiplist: Set[OrderSymbol]) -> "Order":
+        order_type = random.choice(list(OrderType))
+        order_symbol = random.choice(list(set(OrderSymbol) - skiplist))
+        order_quantity = random.randint(MINIMUM_ORDER_QUANTITY, MAXIMUM_ORDER_QUANTITY)
+        return Order(order_type, order_symbol, order_quantity)
+
+
+def prime_match(
+    bank_order_types,
+    bank_order_symbols,
+    bank_order_quantities,
+    client_order_types,
+    client_order_symbols,
+    client_order_quantities,
+):
+    with fhe.tag("comparing-order-types"):
+        types_differ = bank_order_types.reshape(-1, 1) != client_order_types
+
+    with fhe.tag("comparing-order-symbols"):
+        symbols_match = bank_order_symbols.reshape(-1, 1) == client_order_symbols
+
+    with fhe.tag("checking-if-order-can-be-filled"):
+        can_fill = symbols_match & types_differ
+
+    with fhe.tag("calculating-matching-order-quantity"):
+        matching_quantity = np.minimum(
+            bank_order_quantities.reshape(-1, 1), client_order_quantities
+        )
+
+    with fhe.tag("calculating-filled-order-quantity"):
+        filled_quantity = fhe.multivariate(lambda x, y: x * y)(can_fill, matching_quantity)
+
+    bank_result = (
+        bank_order_types,
+        bank_order_symbols,
+        np.sum(filled_quantity, axis=1),
+    )
+
+    client_result = (
+        client_order_types,
+        client_order_symbols,
+        np.sum(filled_quantity, axis=0),
+    )
+
+    return *bank_result, *client_result
+
+
+inputset = [
+    (
+        np.random.randint(0, len(OrderType), size=(NUMBER_OF_BANK_ORDERS,)),
+        np.array(
+            [int(symbol) for symbol in random.sample(list(OrderSymbol), NUMBER_OF_BANK_ORDERS)]
+        ),
+        np.random.randint(
+            MINIMUM_ORDER_QUANTITY,
+            MAXIMUM_ORDER_QUANTITY + 1,
+            size=(NUMBER_OF_BANK_ORDERS,),
+        ),
+        np.random.randint(0, len(OrderType), size=(NUMBER_OF_CLIENT_ORDERS,)),
+        np.array(
+            [int(symbol) for symbol in random.sample(list(OrderSymbol), NUMBER_OF_CLIENT_ORDERS)]
+        ),
+        np.random.randint(
+            MINIMUM_ORDER_QUANTITY,
+            MAXIMUM_ORDER_QUANTITY + 1,
+            size=(NUMBER_OF_CLIENT_ORDERS,),
+        ),
+    )
+    for _ in range(1000)
+]
+
+# comparison_strategy_preference will be used for == and !=
+# bitwise_strategy_preference will be used for &
+# multivariate_strategy_preference will be used for the multivariate function
+# min_max_strategy_preference will be used for np.minimum
+configuration = fhe.Configuration(
+    enable_unsafe_features=True,
+    use_insecure_key_cache=True,
+    insecure_key_cache_location=".keys",
+    fhe_simulation=True,
+    comparison_strategy_preference=fhe.ComparisonStrategy.ONE_TLU_PROMOTED,
```

Comment:
I still believe this example is good for illustrating these strategies, maybe a followup could be to link the docs with variants of this example to see the effect of switching strategies?

Reply:
I added a 

```
# Changing these strategies and compiling with show_mlir=True can be a didactic
# way to understand the impact of the different strategies, on the produced
# MLIR and on speed. Remark that a given strategy may be optimal for given
# input bitwidth and not optimal for other bitwidths (so changing the number of
# order symbols will change the optimal choice). Look into
# http://docs.zama.ai/concrete for more precision about the strategies.
```

but I don't feel like doing a tutorial on the strategies. 

I can't really link to a precise place in the doc, I added something more generic.